prov/util/src/util_attr.c

```diff
@@ -372,10 +372,22 @@ int fi_check_domain_attr(const struct fi_provider *prov,
 		return -FI_ENODATA;
 	}
 
+	if (user_attr->caps & ~(prov_attr->caps)) {
+		FI_INFO(prov, FI_LOG_CORE, "Requested domain caps not supported\n");
+		FI_INFO_CAPS(prov, prov_attr, user_attr, caps, FI_TYPE_CAPS);
+		return -FI_ENODATA;
+	}
+
+	if ((user_attr->mode & prov_attr->mode) != prov_attr->mode) {
+		FI_INFO(prov, FI_LOG_CORE, "Invalid domain mode\n");
```

Comment:
nit: I would reword this message to something more like "Require domain mode missing"